File: hashmap/1-Two_Sum/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        value_map = {}
        for i in range(len(nums)):
            value = nums[i]
            residus = target - value
            if residus in value_map:
                return [value_map[residus], i]
            else:
                value_map[value] = i
        return []

File: hashmap/1-Two_Sum/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> valueMap;
        int residus;
        for (int i = 0; i < nums.size(); i++) {
            residus = target - nums[i];
            if (valueMap.find(residus) != valueMap.end()) {
                result.push_back(valueMap[residus]);
                result.push_back(i);
                return result;
            } else {
                valueMap[nums[i]] = i;
            }
        }

        return result;
    }
};
